### ISSUE TITLE
OAuth with client/services pattern

### ISSUE THREAD

**Issue by andrews05** [stripe/stripe-php#958]
I'm using the new client/services pattern (StripeClient) but as far as I can tell, the OAuth doesn't support this pattern yet, which means I still have to do this to get a token:
```PHP
Stripe::setApiKey($apiKey);
$response = OAuth::token([
    'grant_type' => 'authorization_code',
    'code' => $_REQUEST['code']
]);
```
Will this be updated to work with the StripeClient?

**Comment by remi-stripe** [comment]
Thanks for the report @andrews05. At first we wanted to focus on the API layer from api.stripe.com for the client so we have not immediately implemented OAuth but I'll tag this as a future change and flag internally.

**Comment by andrews05** [comment]
Sure, thank you :)

**Comment by maxleroy** [comment]
Thank you guys it's a real pleasure to see this new implementation coming to life.
I'd need the OAuth feature to be implemented ASAP as well.
Do you have any idea when we can expect it ?
Again thank you so much 🙏

**Comment by remi-stripe** [comment]
@maxleroy I don't have a timeline for this change I'm sorry. This shouldn't be an emergency since it shouldn't impact your integration here. Can you give me a bit more details on what this would change on your end and why you consider it urgent?

**Comment by richardm-stripe** [comment]
Released in [v7.40.0](https://github.com/stripe/stripe-php/releases/tag/v7.40.0) e.g.
```php
$stripe->oauth->token([
    'grant_type' => 'authorization_code',
    'code' => $_REQUEST['code']
]);
```
